core/server/worker/src/test/java/alluxio/worker/block/MonoBlockStoreCommitBlockTest.java

```
@@ -0,0 +1,142 @@
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+        mTieredBlockStore = new TieredBlockStore(mBlockMetadataManager, mBlockLockManager);
+
+        prepareBlockStore();
+
+        assertThrows(RuntimeException.class, () -> {
+            mMonoBlockStore.commitBlock(sessionId, blockId, false);
+        });
+
+        verify(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+        verify(mListener, never()).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+    }
+
+
+    @Test
+    public void CommitLocalFailandCommitMasterSuccess() {
+        mTieredBlockStore = spy(new TieredBlockStore(mBlockMetadataManager, mBlockLockManager));
+        doAnswer((i) -> {
+            throw new RuntimeException();
+        }).when(mTieredBlockStore).commitBlockInternal(anyLong(), anyLong(), anyBoolean());
+
+        prepareBlockStore();
+
+        assertThrows(RuntimeException.class, () -> {
+            mMonoBlockStore.commitBlock(sessionId, blockId, false);
+        });
+
+        verify(mListener, never()).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+        verify(mListener, never()).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+    }
+
+    public void prepareBlockStore() {
+        mMonoBlockStore = new MonoBlockStore(mTieredBlockStore, mockedBlockMasterClientPool, mock(UfsManager.class), new AtomicReference<>(1L));
+
+        try {
+            TieredBlockStoreTestUtils.createTempBlock(sessionId, blockId, 64, mTestDir1);
+        } catch (Exception e) {
+            System.out.println("createTempBlockExceptiona: " + e);
+        }
+
+        byte[] data = new byte[64];
+        Arrays.fill(data, (byte) 1);
+        ByteBuffer buf = ByteBuffer.wrap(data);
+        try (BlockWriter writer = mMonoBlockStore.createBlockWriter(sessionId, blockId)) {
+            writer.append(buf);
+        } catch (Exception e) {
+            System.out.println(e);
+        }
```

no need to suppress the exception here: the writing is expected to succeed, otherwise the test should simply fail so that you would notice. proceeding the test with the error does not make sense.